packages/expo-camera/ios/EXCamera/EXCamera.m

```diff
@@ -413,60 +413,58 @@ - (void)captureOutput:(AVCapturePhotoOutput *)output didFinishProcessingPhoto:(A
+  [self handleCapturedImageData:imageData exifMetadata:photo.metadata[(NSString *)kCGImagePropertyExifDictionary] options:options resolver:resolve reject:reject];
 }
 
-- (void)handleCapturedImageData:(NSData *)imageData exifMetadata:(NSDictionary *)exifMetadata options:(NSDictionary *)options resolver:(UMPromiseResolveBlock)resolve
+- (void)handleCapturedImageData:(NSData *)imageData exifMetadata:(NSDictionary *)exifMetadata options:(NSDictionary *)options resolver:(UMPromiseResolveBlock)resolve reject:(UMPromiseRejectBlock)reject
 {
   UIImage *takenImage = [UIImage imageWithData:imageData];
   BOOL useFastMode = [options[@"fastMode"] boolValue];
   if (useFastMode) {
     resolve(nil);
   }
 
-  CGImageRef takenCGImage = takenImage.CGImage;
-
   CGSize previewSize;
   if (UIInterfaceOrientationIsPortrait([[UIApplication sharedApplication] statusBarOrientation])) {
     previewSize = CGSizeMake(self.previewLayer.frame.size.height, self.previewLayer.frame.size.width);
   } else {
     previewSize = CGSizeMake(self.previewLayer.frame.size.width, self.previewLayer.frame.size.height);
   }
-
+  
+  CGImageRef takenCGImage = takenImage.CGImage;
   CGRect cropRect = CGRectMake(0, 0, CGImageGetWidth(takenCGImage), CGImageGetHeight(takenCGImage));
   CGRect croppedSize = AVMakeRectWithAspectRatioInsideRect(previewSize, cropRect);
   takenImage = [EXCameraUtils cropImage:takenImage toRect:croppedSize];
 
-  float quality = [options[@"quality"] floatValue];
-  NSData *takenImageData = UIImageJPEGRepresentation(takenImage, quality);
-
   NSString *path = [self.fileSystem generatePathInDirectory:[self.fileSystem.cachesDirectory stringByAppendingPathComponent:@"Camera"] withExtension:@".jpg"];
+  float width = takenImage.size.width;
+  float height = takenImage.size.height;
+  NSData *processedImageData = nil;
+  float quality = [options[@"quality"] floatValue];
 
   NSMutableDictionary *response = [[NSMutableDictionary alloc] init];
-  response[@"uri"] = [EXCameraUtils writeImage:takenImageData toPath:path];
-  response[@"width"] = @(takenImage.size.width);
-  response[@"height"] = @(takenImage.size.height);
-
-  if ([options[@"base64"] boolValue]) {
-    response[@"base64"] = [takenImageData base64EncodedStringWithOptions:0];
+  if ([options[@"exif"] boolValue]) {
+    NSMutableDictionary *updatedExif = [EXCameraUtils updateExifMetadata:exifMetadata withAdditionalData:@{ @"Orientation": @([EXCameraUtils exportImageOrientation:takenImage.imageOrientation]) }]; // TODO
+    updatedExif[(NSString *)kCGImagePropertyExifPixelYDimension] = @(width);
+    updatedExif[(NSString *)kCGImagePropertyExifPixelXDimension] = @(height);
+    response[@"exif"] = updatedExif;
+    
+    // UIImage does not contain metadata information. We need to add them to CGImage manually.
+    processedImageData = [EXCameraUtils writeImageToData:takenImage withExifData:updatedExif imageQuality:quality];
```

Comment:
```suggestion
    processedImageData = [EXCameraUtils dataFromImage:takenImage withExifData:updatedExif imageQuality:quality];
```
Method name suggests other process IMO (taking image and writing it to some data container).